app/navigation/NavigatorForChangeActivity.scala

```diff
@@ -119,34 +120,53 @@ class NavigatorForChangeActivity @Inject() extends Navigator {
     case _ => _ => defaultCall
   }
 
-  private def navigationFollowingImports(userAnswers: UserAnswers): Call = {
-    userAnswers.get(AmountProducedPage) match {
-      case Some(AmountProduced.Large) => navigateForLargeAmountProducedFollowingImports(userAnswers)
-      case Some(AmountProduced.None) => navigateForAmountProducedNoneFollowingImports(userAnswers)
-      case Some(AmountProduced.Small) => defaultCall
+  private def navigationFollowingImports(userAnswers: UserAnswers, mode: Mode): Call = {
+    (userAnswers.get(AmountProducedPage), mode) match {
+      case (Some(AmountProduced.Large), mode) => navigateForLargeAmountProducedFollowingImports(userAnswers, mode)
+      case (Some(AmountProduced.None ), mode) => navigateForAmountProducedNoneFollowingImports(userAnswers, mode)
+      case (Some(AmountProduced.Small), mode) => navigateForAmountProducedSmallProducerImports(userAnswers, mode)
       case _ => routes.AmountProducedController.onPageLoad(NormalMode)
     }
   }
 
-  private def navigateForAmountProducedNoneFollowingImports(userAnswers: UserAnswers): Call =
-    userAnswers.get(ContractPackingPage)
+  private def navigateForAmountProducedSmallProducerImports(userAnswers: UserAnswers, mode: Mode): Call = {
+    (userAnswers.get(SecondaryWarehouseDetailsPage), mode)
     match {
-      case Some(true) if userAnswers.packagingSiteList.isEmpty =>
+      case (Some(_), CheckMode) => routes.ChangeActivityCYAController.onPageLoad
+      case (_ ,  CheckMode)  => routes.SecondaryWarehouseDetailsController.onPageLoad
+      case (_, _) => defaultCall
+    }
+  }
+
+  private def navigateForAmountProducedNoneFollowingImports(userAnswers: UserAnswers, mode: Mode): Call =
+    (userAnswers.get(ContractPackingPage), userAnswers.get(SecondaryWarehouseDetailsPage), mode)
+    match {
+      case (Some(true), _, NormalMode) if userAnswers.packagingSiteList.isEmpty =>
         routes.PackAtBusinessAddressController.onPageLoad(NormalMode)
-      case Some(true) =>
+      case (Some(true), _, NormalMode) =>
         routes.PackagingSiteDetailsController.onPageLoad(NormalMode)
-      case Some(false) => routes.SecondaryWarehouseDetailsController.onPageLoad
-      case _ => routes.SecondaryWarehouseDetailsController.onPageLoad
+      case (Some(false), _, NormalMode) =>
+        routes.SecondaryWarehouseDetailsController.onPageLoad
+      case (_, _ , NormalMode) =>
+        routes.SecondaryWarehouseDetailsController.onPageLoad
+      case (_, Some(_), CheckMode) =>
```

Comment:
I'd probably swap these this case and the one below around and check for None. Happy to leave it if you don't think that is neater?